src/encoded/viewconfigs/matrix.py

```diff
@@ -202,50 +196,60 @@ def preprocess_view(self):
-        matrix_route = self._request.route_path('matrix', slash='/')
-        self._matrix['clear_matrix'] = matrix_route + '?type=' + self._doc_types[0]
-        self._result['views'] = self._construct_result_views(type_info)
-        query, used_filters = self._construct_query()
-        es_results = self._elastic_search.search(body=query, index=self._es_index)
+        matrix['search_base'] = search_route + self._search_base
+        matrix['clear_matrix'] = matrix_route + '?type=' + self._doc_types[0]
+        search_result = {
+            '@context': self._request.route_path('jsonld_context'),
+            'filters': [],
+            '@id': matrix_route + self._search_base,
+            '@type': ['Matrix'],
+            'notification': '',
+            'title': self._set_result_title(type_info),
+            'matrix': matrix,
+            'view': self._construct_result_views(type_info),
+        }
+        search_query, used_filters = self._construct_query(search_result, matrix)
+        es_results = self._elastic_search.search(
+            body=search_query,
+            index=self._es_index
+        )
         aggregations = es_results['aggregations']
         total = aggregations['matrix']['doc_count']
-        self._result['matrix']['doc_count'] = total
-        self._result['matrix']['max_cell_doc_count'] = 0
-        self._result['facets'] = self._format_facets(
+        search_result['matrix']['doc_count'] = total
+        search_result['matrix']['max_cell_doc_count'] = 0
+        search_result['facets'] = self._format_facets(
             es_results,
             self._facets,
             used_filters,
             (self._schema,),
             total,
             self._principals
         )
-        x_grouping = self._matrix['x']['group_by']
-        y_groupings = self._matrix['y']['group_by']
+        x_grouping = matrix['x']['group_by']
+        y_groupings = matrix['y']['group_by']
         self._summarize_buckets(
             aggregations['matrix']['x']['buckets'],
             aggregations['matrix'],
-            y_groupings + [x_grouping]
+            y_groupings + [x_grouping],
+            matrix
         )
-        self._result['matrix']['y'][y_groupings[0]] = aggregations['matrix'][y_groupings[0]]
-        self._result['matrix']['x'].update(aggregations['matrix']['x'])
-        self._result.update(
+        search_result['matrix']['y'][y_groupings[0]] = aggregations['matrix'][y_groupings[0]]
+        search_result['matrix']['x'].update(aggregations['matrix']['x'])
+        search_result.update(
             search_result_actions(
                 self._request,
                 self._doc_types,
                 es_results
             )
         )
-        self._result['total'] = es_results['hits']['total']
-        if self._result['total']:
-            self._result['notification'] = 'Success'
+        search_result['total'] = es_results['hits']['total']
```

Comment:
Now that we've collected everything we need to make search_results, build it here with remaining keys.